### PR TITLE
Add ability to import owned tokens to resolution-swift library

### PR DESCRIPTION
**Pull request by JohnnyJumper**
Introduce Resolution#tokensOwnedBy(address: String, service: String) throws -> [String]

Cns#tokensOwnedBy fetches the registry contract for logs on the provided address, fetches NewURI to decrypt the domains
Zns is not supported at the moment.
https://www.pivotaltracker.com/story/show/178446775

### CODE REVIEW

**Review by merenkoff — APPROVED** [review]
LGTM

**Comment by bogdan** [comment]
@rommex please take a look also

File: Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift [hunk 1]
```diff
@@ -91,6 +91,42 @@ internal class CNS: CommonNamingService, NamingService {
         }
     }
 
+    func tokensOwnedBy(address: String) throws -> [String] {
+        let registryContract = try self.buildContract(address: self.contracts.registryAddress, type: .registry)
+        let transferLogs = try registryContract.callLogs(
+            fromBlock: "0x8A958B",
```
Comment (rommex) [inline Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift:97]: Let's extract this value as a constant.
Also let's keep in mind that for testnet - Rinkeby - this value is lower: `0x7232BC`

File: Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift [hunk 2]
```diff
@@ -91,6 +91,42 @@ internal class CNS: CommonNamingService, NamingService {
...
+            signatureHash: "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+            for: address.normalized32,
+            isTransfer: true
+        ).compactMap {
+            $0.topics[3]
+        }
+
+        let domainsData = try transferLogs.compactMap {
+            try registryContract.callLogs(
+                fromBlock: "0x8A958b",
```
Comment (rommex) [inline Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift:107]: +

File: Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift [hunk 3]
```diff
@@ -91,6 +91,42 @@ internal class CNS: CommonNamingService, NamingService {
         }
     }
 
+    func tokensOwnedBy(address: String) throws -> [String] {
+        let registryContract = try self.buildContract(address: self.contracts.registryAddress, type: .registry)
+        let transferLogs = try registryContract.callLogs(
+            fromBlock: "0x8A958B",
+            signatureHash: "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
```
Comment (rommex) [inline Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift:98]: I suggest we extract this Transfer Event hash as  a public accessible constant

File: Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift [hunk 4]
```diff
@@ -91,6 +91,42 @@ internal class CNS: CommonNamingService, NamingService {
...
+            for: address.normalized32,
+            isTransfer: true
+        ).compactMap {
+            $0.topics[3]
+        }
+
+        let domainsData = try transferLogs.compactMap {
+            try registryContract.callLogs(
+                fromBlock: "0x8A958b",
+                signatureHash: "0xc5beef08f693b11c316c0c8394a377a0033c9cf701b8cd8afd79cecef60c3952",
```
Comment (rommex) [inline Sources/UnstoppableDomainsResolution/NamingServices/CNS.swift:108]: As well for the NewURI hash

**Review by rommex — CHANGES_REQUESTED** [review]

**Comment by JohnnyJumper** [comment]
By default we are using this key for swift library: 3c25f57353234b1b853e9861050f4817
We need to whitelist the eth_getLogs method on infura for this key. Otherwise, the default settings will not work for users.

**Review by rommex — APPROVED** [review]
Cool

**Comment by DeRain** [comment]
> By default we are using this key for swift library: 3c25f57353234b1b853e9861050f4817
> We need to whitelist the eth_getLogs method on infura for this key. Otherwise, the default settings will not work for users.

Done, thank you! 👍

**Review by DeRain — APPROVED** [review]
👍

**Pull request merged**